platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/daemon/ProgramExecutionDaemon.java

```diff
@@ -107,18 +111,22 @@ private void addRunnable(AGProgram program) {
 
 	private void terminateProgram(AGProgram program) {
 
-		terminate(getRunningRunnableThreads(program));
-		program.resetState();
+		if (terminate(getRunningRunnableThreads(program))) {
+			program.resetState();
+		}
```

Comment:
This is the actual change, that is, don't reset the state if execution was not terminated.